Tests/SwiftSyntaxBuilderTest/StructTests.swift

```diff
@@ -30,7 +30,7 @@ final class StructTests: XCTestCase {
       genericWhereClause: GenericWhereClause {
         GenericRequirement(body: ConformanceRequirement(leftTypeIdentifier: "A", rightTypeIdentifier: "X"))
         GenericRequirement(body: SameTypeRequirement(
-            leftTypeIdentifier: "A.P", equalityToken: .identifier("=="), rightTypeIdentifier: "D"))
+            leftTypeIdentifier: "A.P", equalityToken: "==", rightTypeIdentifier: "D"))
```

Comment:
I think this is mostly what I’m worried about. Technically, `==` should never have been an `identifier` but `.spacedBinaryOperator("==")`. This doesn’t really matter if all you do with the syntax tree is print it but e.g. a formatter might care about the distinction. 

It’s not the most pretty solution but what do you think about the following: We introduce a new type `IdentifierToken` and use that as a parameter for all syntax children that have type `IdentifierToken`
```swift
struct IdentifierToken: ExpressibleByStringLiteral {
  let token: TokenSyntax

  init(_ token: TokenSyntax) {
    assert(token.tokenKind == .identifier)
    self.token = token
  }
  
  // If necessary, also add these
  func withLeadingTrivia(_ leadingTrivia: Trivia) -> IdentifierToken { /* … */ }
  func withTrailingTrivia(_ trailing: Trivia) -> IdentifierToken { /* … */ }
}
```

That would cover the common case of just passing in a string for the identifier. Using any of the functions that are defined on `TokenSyntax` becomes a little harder because you first need to create `TokenSyntax.identifier`, then modify it and wrap it in `IdentifierToken`, but maybe that’s not a big issue in practice.

Reply:
Oh, yeah, we should fix that test.

I like the idea of statically encoding where we expect an identifier. The only thing I could think about is that there are a few places (like argument labels) where also other tokens like `.wildcard` are 'allowed', but perhaps it would be fine to keep `TokenSyntax` in these places to make this explicit at the use site.

Reply:
So I have experimented a bit with this idea [over here](https://github.com/fwcd/swift-syntax/tree/identifier-tokens), unfortunately fewer nodes than I expected are pure/arbitrary identifiers. Most prominently, `FunctionDecl` still has to take a `TokenSyntax` since it could e.g. represent a operator too. Since the vast majority of uses (at least in the test suite and among the `SwiftSyntaxBuilderGeneration` templates is `FunctionDecl` identifiers, I don't think such a workaround would be worth the hassle for now.

Perhaps we could just add a convenience initializer for `FunctionDecl` that takes a `String` parameter? It feels like there ought to be some more general solution, but apart from providing an `ExpressibleAsStringLiteral` conformance on `TokenSyntax` directly, I don't really see how we could provide this sugar without disproportionally complicating some other part of `SwiftSyntaxBuilder`.